core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -150,6 +153,44 @@ protected void validate(TableMetadata currentMetadata) {
    */
   protected abstract List<ManifestFile> apply(TableMetadata metadataToUpdate);
 
+  /**
+  Implementation of this method update stats for each partition and
+  returns Map which defines partition specification for which udpates happen
+  Map Key: Partition Specification Id Map Value: New location of partition Stats file
+   */
+  protected Map<Integer, String> updatePartitionStats() {
```

Comment:
This looks wierd, why is an update method not doing anything?

Reply:
Defined this method as  "abstract". Any implementation of updatePartitionStats method in SnapshotProducer.java which does not want to updated stats. needs to return empty map.